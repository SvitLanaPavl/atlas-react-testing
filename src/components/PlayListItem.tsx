import React from 'react'

const PlayListItem = ( {title, artist, duration, isCurrent} ) => {
  return (
    <button className={`flex w-full justify-between items-center mb-1 cursor-pointer ${isCurrent ? 'bg-customPurple-50 rounded-md' : 'bg-transparent'} hover:bg-customPurple-50 transition duration-150 ease-in-out rounded-md p-1`}>
      <div className='text-left'>
        <div className='font-medium  text-customPurple-700'>{title}</div>
        <div className='text-sm leading-5 text-customPurple-300'>{artist}</div>
      </div>
      <div className='text-sm leading-5 text-customPurple-300'>{duration}</div>
    </button>
  );
}

export default PlayListItem