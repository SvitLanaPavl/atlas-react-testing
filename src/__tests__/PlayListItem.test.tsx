import { render } from '@testing-library/react';
import { expect, test, vi } from 'vitest';
import PlayListItem from '@/components/PlayListItem';

const onClick = vi.fn();

test('Playlist renders correctly when current', () => {
  const { container } = render(
    <PlayListItem title='Painted in Blue' artist='Soul Canvas' duration='5:55' isCurrent={true} onClick={onClick} />
  );
  expect(container).toMatchSnapshot();
});

test('Playlist renders correctly when not current', () => {
  const { container } = render(
    <PlayListItem title='Tidal Drift' artist='Echoes of the Sea' duration='8:02' isCurrent={false} onClick={onClick} />
  );
  expect(container).toMatchSnapshot();
});