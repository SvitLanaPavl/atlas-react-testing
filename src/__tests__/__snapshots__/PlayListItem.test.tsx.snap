// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Playlist renders correctly when current 1`] = `
<div>
  <button
    class="flex w-full justify-between items-center mb-1 cursor-pointer bg-customPurple-50 rounded-md hover:bg-customPurple-50 transition duration-150 ease-in-out rounded-md p-1"
  >
    <div
      class="text-left"
    >
      <div
        class="font-medium  text-customPurple-700"
      >
        Painted in Blue
      </div>
      <div
        class="text-sm leading-5 text-customPurple-300"
      >
        Soul Canvas
      </div>
    </div>
    <div
      class="text-sm leading-5 text-customPurple-300"
    >
      5:55
    </div>
  </button>
</div>
`;

exports[`Playlist renders correctly when not current 1`] = `
<div>
  <button
    class="flex w-full justify-between items-center mb-1 cursor-pointer bg-transparent hover:bg-customPurple-50 transition duration-150 ease-in-out rounded-md p-1"
  >
    <div
      class="text-left"
    >
      <div
        class="font-medium  text-customPurple-700"
      >
        Tidal Drift
      </div>
      <div
        class="text-sm leading-5 text-customPurple-300"
      >
        Echoes of the Sea
      </div>
    </div>
    <div
      class="text-sm leading-5 text-customPurple-300"
    >
      8:02
    </div>
  </button>
</div>
`;
